src/player/player.gd

```diff
@@ -48,8 +53,14 @@ func _move(_input_dir:Vector2) -> void:
 
 	if next_tile != null:
 		print(next_tile)
+		if !has_boots and next_tile.is_in_group('chest'):
+			var new_obtained_boots = boots_packed_scene.instantiate()
+			self.add_child(new_obtained_boots)
+			has_boots = true
+			pass
 	else:
 		position += _input_dir * TILE_SIZE
+		emit_signal("made_successful_move", _input_dir)
```

Comment:
please use godot 4 callable function instead of the godot 3 string passing function
should be 
made_successful_move.emit(_input_dir)